projects/plugins/jetpack/modules/subscriptions/subscribe-modal/class-jetpack-subscribe-modal.php

```diff
@@ -0,0 +1,236 @@
+<?php
+/**
+ * Adds support for Jetpack Subscribe Modal feature
+ *
+ * @package automattic/jetpack-mu-wpcom
+ * @since $$next-version$$
+ */
+
+/**
+ * Jetpack_Subscribe_Modal class.
+ */
+class Jetpack_Subscribe_Modal {
+	/**
+	 * Jetpack_Subscribe_Modal singleton instance.
+	 *
+	 * @var Jetpack_Subscribe_Modal|null
+	 */
+	private static $instance;
+
+	/**
+	 * Jetpack_Subscribe_Modal instance init.
+	 */
+	public static function init() {
+		if ( self::$instance === null ) {
+			self::$instance = new Jetpack_Subscribe_Modal();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Jetpack_Subscribe_Modal class constructor.
+	 */
+	public function __construct() {
+		add_action( 'wp_loaded', array( $this, 'wpcom_create_subscribe_template' ) );
+		add_action( 'wp_footer', array( $this, 'wpcom_add_subscribe_modal_to_frontend' ) );
+		add_action( 'wp_enqueue_scripts', array( $this, 'wpcom_enqueue_subscribe_modal_assets' ) );
+	}
+
+	/**
+	 * Adds a Subscribe Modal template part that can be edited in the editor.
+	 * This is later loaded in a pop up modal for Newsletter sites.
+	 *
+	 * @return void
+	 */
+	public function wpcom_create_subscribe_template() {
+		if ( $this->should_enable_subscriber_modal() ) {
+			$post = get_page_by_path( $this->get_subscribe_template_slug(), OBJECT, 'wp_template_part' );
+
+			if ( ! $post ) {
+				$template = array(
+					'slug'         => $this->get_subscribe_template_slug(),
+					'post_name'    => $this->get_subscribe_template_slug(),
+					'post_title'   => __( 'Subscribe Modal', 'jetpack' ),
+					'post_content' => $this->get_subscribe_template_content(),
+					'post_status'  => 'publish',
+					'post_author'  => 1,
+					'post_type'    => 'wp_template_part',
+					'scope'        => array(),
+					'tax_input'    => array(
+						'wp_theme' => get_option( 'stylesheet' ),
+					),
+				);
+				wp_insert_post( $template );
+			}
+		}
+	}
+
+	/**
+	 * Adds modal with Subscribe Modal content.
+	 *
+	 * @return void
+	 */
+	public function wpcom_add_subscribe_modal_to_frontend() {
+		if (
+			$this->should_enable_subscriber_modal() &&
+			! is_admin() &&
+			is_single() &&
+			'post' == get_post_type()
+		) {
+			$posts = get_posts(
+				array(
+					'post_type'   => 'wp_template_part',
+					'post_status' => 'publish',
+					'numberposts' => -1,
```

Comment:
I think you can pass 

```php
`name` => $this->get_subscribe_template_slug(),
```

Reply:
Good call @lezama! Updated this. While I was at it, I also added a check to confirm we found posts in this query before we proceed to try adding the template modal.